### PR TITLE
docs: add stdlib env::var(_os) panic

### LINKED ISSUE #63456: std::env::var and std::env::var_os do not document panic on key containing null byte
`env::remove_var` and `env::set_var` document that they panic when the key contains a null byte, but `env::var` and `env::var_os` do not document this behavior.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e43934f459b10424f374f20db846f092

### PR DESCRIPTION
**Pull request by tommilligan**
Closes #63456

### CODE REVIEW

File: src/libstd/env.rs [hunk 1]
```diff
@@ -210,6 +216,12 @@ fn _var(key: &OsStr) -> Result<String, VarError> {
 ///
 /// [`None`]: ../option/enum.Option.html#variant.None
 ///
+/// # Panics
+///
+/// This function may panic if `key` is empty, contains an ASCII equals sign
+/// `'='` or the NUL character `'\0'`, or when the value contains the NUL
+/// character.
```
Comment (Mark-Simulacrum) [inline src/libstd/env.rs]: NUL character is repeated twice in both of these?

Comment (tommilligan) [reply src/libstd/env.rs]: This is copied verbatim from the underlying functions called, which state they may crash when either the key or the value contain a null byte.

Comment (Mark-Simulacrum) [reply src/libstd/env.rs]: Oh, I missed the key/value distinction.

**Comment by joshtriplett** [comment]
This looks great to me.

@bors r+ rollup

File: src/libstd/env.rs [hunk 2]
```diff
@@ -182,6 +182,12 @@ impl fmt::Debug for VarsOs {
 /// * Environment variable is not present
 /// * Environment variable is not valid unicode
 ///
+/// # Panics
+///
+/// This function may panic if `key` is empty, contains an ASCII equals sign
```
Comment (bluetech) [inline src/libstd/env.rs]: Is it "may panic" or "panics" (definitive)?

Comment (tommilligan) [reply src/libstd/env.rs]: The function relies on `std::sys::os`, so the behaviour differs by operating system. We could update the docs to make this clearer.

Comment (bluetech) [reply src/libstd/env.rs]: Hmm, I'm not sure what is the practice in this case. Maybe "may" is fine.

BTW, I do wonder why this behavior was chosen, that if `key` contains a nul then it panics. It makes sense for *setting* a var, but when getting, it makes more sense to me that it just return `None`. The code comment in the unix implementation seems to agree, I think..

https://github.com/rust-lang/rust/blob/ad7c55e1fc55d9af4787b285cec1c64e3480ae84/src/libstd/sys/unix/os.rs#L469-L471

Comment (bluetech) [inline src/libstd/env.rs]: Another question: the function documentation says that when `key` is not present, an error is returned, not a panic. So is the "is empty" here correct?

**Pull request merged**